extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -652,6 +656,71 @@ export async function newCredential(
     return;
   }
 
+  // ***************************************************************
+  // Step: Select authentication method (Connect only)
+  // ***************************************************************
+  async function inputAuthMethod(input: MultiStepInput, state: MultiStepState) {
+    const authMethods = [
+      {
+        label: "Token Authentication",
+        description: "Recommended - one click connection",
+      },
+      { label: "API Key", description: "Manually enter an API key" },
+    ];
+
+    const pick = await input.showQuickPick({
+      title: state.title,
+      step: 0,
+      totalSteps: 0,
+      placeholder: "Select authentication method",
+      items: authMethods,
+      activeItem: authMethods[0], // Token authentication is default
+      buttons: [],
+      shouldResume: () => Promise.resolve(false),
+      ignoreFocusOut: true,
+    });
+
+    state.data.authMethod = pick.label === "API Key" ? "apiKey" : "token";
+
+    if (state.data.authMethod === "apiKey") {
+      return {
+        step: (input: MultiStepInput) => inputAPIKey(input, state),
+      };
+    } else {
+      return {
+        step: (input: MultiStepInput) => inputToken(input, state),
+      };
+    }
+  }
+
+  // ***************************************************************
+  // Step: Generate and claim token (Connect only)
+  // ***************************************************************
+  async function inputToken(_input: MultiStepInput, state: MultiStepState) {
+    // url should always be defined by the time we get to this step
+    const serverUrl = typeof state.data.url === "string" ? state.data.url : "";
+
+    try {
+      // Create and initialize the token activator
+      const tokenActivator = new ConnectAuthTokenActivator(serverUrl, viewId);
+      await tokenActivator.initialize();
+
+      // Activate the token
+      const result = await tokenActivator.activateToken();
```

Comment:
I really like how this turned out, specially because the `activator` now has insurance (tests) 😄